pilot/pkg/model/context.go

```diff
@@ -536,7 +545,7 @@ type NodeMetadata struct {
 	// Labels specifies the set of workload instance (ex: k8s pod) labels associated with this node.
 	Labels map[string]string `json:"LABELS,omitempty"`
```

Comment:
So IIUC, this should ~never be used except for in the initialization. Everything else should use `proxy.Labels` not `proxy.Metadata.Labels`? If so, it seems like we missed a few - EffectiveProxyConfig for example

Reply:
`EffectiveProxyConfig` is called in injector, which is using pod labels, so no influence on it

```
	if wh.env.PushContext != nil && wh.env.PushContext.ProxyConfigs != nil {
		if generatedProxyConfig := wh.env.PushContext.ProxyConfigs.EffectiveProxyConfig(
			&model.NodeMetadata{
				Namespace:   pod.Namespace,
				Labels:      pod.Labels,
				Annotations: pod.Annotations,
			}, wh.meshConfig); generatedProxyConfig != nil {
			proxyConfig = generatedProxyConfig
		}
	}
```